src/pixel-ping.coffee

```diff
@@ -29,22 +29,36 @@ serialize = ->
 
 # Reset the `store`.
 reset = ->
+  old_store = store
   store = {}
+  old_store
+
+# Merge the given `store` with the current one.
+merge = (new_store) ->
+  for key, hits of new_store
+    store[key] or= 0
+    store[key] +=  hits
 
 # Flushes the `store` to be saved by an external API. The contents of the store
 # are sent to the configured `endpoint` URL via HTTP POST. If no `endpoint` is
 # configured, this is a no-op.
 flush = ->
   log store
   return unless config.endpoint
+  on_error = (message) ->
+    if !config.discard
+      merge(old_store)
+    console.log message
   data = serialize()
+  old_store = reset()
   endReqOpts['headers']['Content-Length'] = data.length
-  request = http.request endReqOpts, (response) ->
-    reset()
-    console.info '--- flushed ---'
+  request = http.request endReqOpts, (res) ->
+    if res.statusCode <= 299 and res.statusCode >= 200
```

Comment:
This could be nicely abstracted into an `isSuccessStatus` function (or something similar).

Reply:
Good point, fixed!